src/handlers/int-test-support/helpers/call-wrappers.test.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  callWithRetry,
+  callWithRetryAndTimeout,
+  callWithTimeout,
+} from "./call-wrappers";
+
+describe("callWithTimeout", () => {
+  const wait = async (milliseconds: number): Promise<string> => {
+    return await new Promise((resolve) =>
+      setTimeout(() => resolve("a"), milliseconds)
+    );
+  };
+
+  const oneSecondCall = async (): Promise<string> => await wait(1000);
+
+  describe("given a call that takes shorter than the timeout", () => {
+    it("does not fail with a timeout error", async () => {
+      // give it a second and a half to complete
+      const result = await callWithTimeout(1500)(oneSecondCall)();
+      expect(result).toBe("a");
+    });
+  });
+
+  describe("given a call that takes longer than the timeout", () => {
+    it("fails with a timeout error", async () => {
+      try {
+        // only give it a half a second to complete
+        await callWithTimeout(500)(oneSecondCall)();
+      } catch (error) {
+        expect(error).toBeInstanceOf(Error);
+        expect((error as Error).message).toBe("Operation timed out");
+      }
+      expect.hasAssertions();
```

Comment:
Why not use expect:
```suggestion
      await expect(callWithTimeout(500)(oneSecondCall)())
        .rejects.toThrowError(new Error("Operation timed out"));
```

Reply:
Implemented that change through the file now with a bit more extra cleanup.